rkt/fetch.go

```diff
@@ -203,43 +246,61 @@ func download(aciURL string, ascURL string, ds *cas.Store, ks *keystore.Keystore
 		return nil, aciFile, nil
 	}
 
-	var sigTempFile *os.File
-	if ks != nil {
-		stdout("Downloading signature from %v\n", ascURL)
-		sigTempFile, err = downloadSignatureFile(ascURL)
+	if ks != nil && ascFile == nil {
+		u, err := url.Parse(ascURL)
 		if err != nil {
-			return nil, nil, fmt.Errorf("error downloading the signature file: %v", err)
+			return nil, nil, fmt.Errorf("error parsing ASC url: %v", err)
 		}
-		defer sigTempFile.Close()
-		defer os.Remove(sigTempFile.Name())
+		if u.Scheme != "file" {
+			stdout("Downloading signature from %v\n", ascURL)
+			ascFile, err = downloadSignatureFile(ascURL)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error downloading the signature file: %v", err)
+			}
+			defer os.Remove(ascFile.Name())
+		} else {
+			ascFile, err = os.Open(u.Path)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error opening signature file: %v", err)
+			}
+		}
+		defer ascFile.Close()
 	}
 
-	acif, err := downloadACI(ds, aciURL)
-	if err != nil {
-		return nil, acif, fmt.Errorf("error downloading the aci image: %v", err)
+	var aciFile *os.File
+	if u.Scheme == "file" {
+		aciFile, err = os.Open(u.Path)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error opening aci image: %v", err)
```

Comment:
damn, we have to get away from "aci image" - just "ACI" (or "ACI file" in this case, please)